src/test/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/contact/ContactServiceTest.kt

```diff
@@ -132,6 +137,65 @@ class ContactServiceTest {
     }
   }
 
+  @Nested
+  inner class UpdateContact {
+    private val now = Instant.now(clock)
+    private val yesterday = now.minus(1L, ChronoUnit.DAYS)
+    private val aContactRequest: ContactRequest =
+      ContactRequest(
+        prisonerName = "new-name",
+        prisonId = "new-prison-code",
+        dob = LocalDate.of(1999, 12, 12),
+        prisonNumber = "new-prison-number"
+      )
+
+    private val aContact = Contact(
+      id = 1L,
+      owner = "some-user",
+      name = "some-name",
+      prisonCode = "some-prison-code",
+      dob = LocalDate.of(1990, 1, 1),
+      prisonNumber = "some-prison-number",
+      created = yesterday,
+      updated = yesterday,
+    )
+
+    private val updatedContact = Contact(
+      id = 1L,
+      owner = aContact.owner,
+      name = aContactRequest.prisonerName,
+      prisonCode = aContactRequest.prisonId,
+      dob = aContactRequest.dob,
+      prisonNumber = aContactRequest.prisonNumber,
+      created = yesterday,
+      updated = now,
+    )
+
+    @Test
+    fun `should return saved contact if contact exists`() {
+      given { contactRepository.getContactByOwnerAndId(anyString(), anyLong()) }.willReturn(aContact)
+      given { contactRepository.save(any()) }.willReturn(updatedContact)
+
+      assertThat(contactService.updateContact("some-user", 1L, aContactRequest)).isEqualTo(updatedContact)
+
+      then(contactRepository).should().getContactByOwnerAndId("some-user", 1L)
```

Comment:
Oh, loving the use of the BDD `then` 👍 😄 
Did you copy that from something I'd written in another test class? (I prefer `then` over `verify` as it fits the BDD syntax so much better ... but its all syntactic sugar so it doesnt really matter 😄  )